x/tx/decode/decode.go

```diff
@@ -167,14 +222,25 @@ func (d *Decoder) Decode(txBytes []byte) (*DecodedTx, error) {
 		}
 	}
 
-	return &DecodedTx{
+	decodedTx := &DecodedTx{
 		Messages:                     msgs,
 		DynamicMessages:              dynamicMsgs,
 		Tx:                           theTx,
 		TxRaw:                        &raw,
 		TxBodyHasUnknownNonCriticals: txBodyHasUnknownNonCriticals,
 		Signers:                      signers,
-	}, nil
+	}
+
+	// store the decoded tx in the tx decoder cache
+	d.txCache.Add(cacheKey, cacheEntry{decodedTx: decodedTx, err: err})
+
+	return decodedTx, nil
+}
+
+func (d *Decoder) returnDecodeError(cacheKey string, err error) (*DecodedTx, error) {
+	// store the error in the tx decoder cache
+	d.txCache.Add(cacheKey, cacheEntry{decodedTx: nil, err: err})
```

Comment:
Is there a tx size limit somewhere further up the stack?  Since errors are being cached a malicious actor could flood a node with huge txs (from any code site calls Decode) which would then be giant keys in the LRU, exhausting memory usage.

Reply:
true the issue is we are caching on all possible decodes not only on execution. So this would introduce a dos vector.

Reply:
As far as I know, LRU doesn't set a maximum limit for cache key size. But if keys are very large, it can use a lot of memory and might even cause out-of-memory problems. We should find a solution to handle this situation.